ocp_resources/machine_config_pool.py

```diff
@@ -2,8 +2,78 @@
 
 
 class MachineConfigPool(Resource):
+    """
+    MachineConfigPool object. API reference:
+    https://docs.openshift.com/container-platform/4.12/rest_api/machine_apis/machineconfigpool-machineconfiguration-openshift-io-v1.html
+
+    Args:
+        machine_config_selector (dict): Matching labels/expressions, to determine which MachineConfig objects
+            to apply this MachineConfigPool object.
+            For filtering based on labels, the `matchLabels` dict is used - the same way as it is used in the
+                nodeSelector (see the example of node_selector["matchLabels"] below).
+            For filtering based on expressions, the `matchExpressions` dict is used - the same way as it is used in the
+                nodeSelector (see the example of node_selector["matchExpressions"] below).
+        configuration (dict): Targeted MachineConfig object for the machine config pool, in the following format:
+            {"name": (str), "source": <List of dicts, each representing a MachineConfig resource>}
+        node_selector (dict): Matching dict with supported selector logic, either labels or expressions.
+            matchLabels example:
+            matchLabels:
+                component: <some component>
+                matchExpressions:
+                - { key: tier, operator: In, values: [cache] }
+                - { key: environment, operator: NotIn, values: [dev] }
+            matchExpressions example:
+            matchExpressions:
+            - key: <resource name>/role
+              operator: In
+              values:
+              - value_1
+              - value_2
```

Comment:
```suggestion
        node_selector (dict): Matching dict with supported selector logic, either labels or expressions. See example under `machine_config_selector`
```

Reply:
Because when a user reads the docstring they should have all the info in one place. One param (preferably the first one in the docstring) should hold the explanation and example. The other can refer to it.